### ISSUE TITLE
The reproduced results are lower than those mentioned in the paper

### ISSUE THREAD

**Issue by zhchen9833** [YuHengsss/Trident#6]
Why is my result lower when reproducing than in the paper, for example, the mIOU of VOC21 is 65.29, which is less than the 67.1 in the paper.
I haven't changed any other parameters in the project, so I just followed the steps you gave, such as "python eval.py -- config config/cfd_VOC21. py -- workdir output/VOC21-- sam_define". Should I need to adjust any parameters?

**Comment by YuHengsss** [comment]
Hi, I have clone this repo and reproduce the result which is the same as the reported in the paper. But there is truely something different in the env. So, I have updated the requirements.txt in this repo. Here is the detailed log that I just reproduce, without changing any config.

 python eval.py --config ./configs/cfg_voc21.py --work-dir ./workdir --sam_refine

/home/yuheng/miniconda3/envs/Trident/lib/python3.10/site-packages/torch/utils/cpp_extension.py:28: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.
  from pkg_resources import packaging  # type: ignore[attr-defined]

11/22 08:56:12 - mmengine - INFO - 
------------------------------------------------------------
System environment:
    sys.platform: linux
    Python: 3.10.19 (main, Oct 21 2025, 16:43:05) [GCC 11.2.0]
    CUDA available: True
    numpy_random_seed: 1926513060
    GPU 0,1,2,3: NVIDIA RTX A6000
    CUDA_HOME: /home/yuheng/miniconda3/envs/Trident
    NVCC: Cuda compilation tools, release 11.6, V11.6.124
    GCC: gcc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
    PyTorch: 2.1.0+cu118
    PyTorch compiling details: PyTorch built with:
  - GCC 9.3
  - C++ Version: 201703
  - Intel(R) oneAPI Math Kernel Library Version 2022.2-Product Build 20220804 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v3.1.1 (Git Hash 64f6bcbcbab628e96f33a62c3e975f8535a7bde4)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - LAPACK is enabled (usually provided by MKL)
  - NNPACK is enabled
  - CPU capability usage: AVX512
  - CUDA Runtime 11.8
  - NVCC architecture flags: -gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86;-gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_90,code=sm_90
  - CuDNN 8.7
  - Magma 2.6.1
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=11.8, CUDNN_VERSION=8.7.0, CXX_COMPILER=/opt/rh/devtoolset-9/root/usr/bin/c++, CXX_FLAGS= -D_GLIBCXX_USE_CXX11_ABI=0 -fabi-version=11 -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -DNDEBUG -DUSE_KINETO -DLIBKINETO_NOROCTRACER -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -O2 -fPIC -Wall -Wextra -Werror=return-type -Werror=non-virtual-dtor -Werror=bool-operation -Wnarrowing -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-strict-overflow -Wno-strict-aliasing -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=old-style-cast -Wno-invalid-partial-specialization -Wno-unused-private-field -Wno-aligned-allocation-unavailable -Wno-missing-braces -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Werror=cast-function-type -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_DISABLE_GPU_ASSERTS=ON, TORCH_VERSION=2.1.0, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=1, USE_NNPACK=ON, USE_OPENMP=ON, USE_ROCM=OFF, 

    TorchVision: 0.16.0+cu118
    OpenCV: 4.6.0
    MMEngine: 0.10.1

Runtime environment:
    cudnn_benchmark: True
    mp_cfg: {'mp_start_method': 'fork', 'opencv_num_threads': 0}
    dist_cfg: {'backend': 'nccl'}
    seed: 1926513060
    Distributed launcher: none
    Distributed training: False
    GPU number: 1
------------------------------------------------------------

11/22 08:56:12 - mmengine - INFO - Config:
data_root = '/home/yuheng/datasets/VOCdevkit/VOC2012'
dataset_type = 'PascalVOCDataset'
default_hooks = dict(
    checkpoint=dict(by_epoch=False, interval=2000, type='CheckpointHook'),
    logger=dict(interval=50, log_metric_by_epoch=False, type='LoggerHook'),
    param_scheduler=dict(type='ParamSchedulerHook'),
    sampler_seed=dict(type='DistSamplerSeedHook'),
    timer=dict(type='IterTimerHook'),
    visualization=dict(interval=5, type='SegVisualizationHook'))
default_scope = 'mmseg'
env_cfg = dict(
    cudnn_benchmark=True,
    dist_cfg=dict(backend='nccl'),
    mp_cfg=dict(mp_start_method='fork', opencv_num_threads=0))
launcher = 'none'
load_from = None
log_level = 'INFO'
log_processor = dict(by_epoch=False)
model = dict(
    beta=1.2,
    checkpoint=None,
    clip_type='openai',
    coarse_thresh=0.2,
    cos_fac=1.2,
    minimal_area=225,
    model_type='ViT-B/16',
    name_path='./configs/cls_voc21.txt',
    prob_thd=0.2,
    refine_neg_cos=True,
    sam_ckpt='/home/yuheng/code/Trident/sam_ckpt/sam_vit_b_01ec64.pth',
    sam_iou_thresh=0.8,
    sam_model_type='vit_b',
    sam_refinement=True,
    slide_crop=336,
    slide_stride=224,
    type='Trident',
    vfm_model='dino')
resume = False
sam_ckpt = '/home/yuheng/code/Trident/sam_ckpt/sam_vit_b_01ec64.pth'
sam_refinement = False
test_cfg = dict(type='TestLoop')
test_dataloader = dict(
    batch_size=1,
    dataset=dict(
        ann_file='ImageSets/Segmentation/val.txt',
        data_prefix=dict(
            img_path='JPEGImages', seg_map_path='SegmentationClass'),
        data_root='/home/yuheng/datasets/VOCdevkit/VOC2012',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(keep_ratio=True, scale=(
                2048,
                448,
            ), type='Resize'),
            dict(type='LoadAnnotations'),
            dict(type='PackSegInputs'),
        ],
        type='PascalVOCDataset'),
    num_workers=4,
    persistent_workers=True,
    sampler=dict(shuffle=False, type='DefaultSampler'))
test_evaluator = dict(
    iou_metrics=[
        'mIoU',
    ], type='IoUMetric')
test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(keep_ratio=True, scale=(
        2048,
        448,
    ), type='Resize'),
    dict(type='LoadAnnotations'),
    dict(type='PackSegInputs'),
]
vis_backends = [
    dict(type='LocalVisBackend'),
]
visualizer = dict(
    alpha=1.0,
    name='visualizer',
    type='SegLocalVisualizer',
    vis_backends=[
        dict(type='LocalVisBackend'),
    ])
work_dir = './workdir'

Using cache found in /home/yuheng/.cache/torch/hub/facebookresearch_dino_main
11/22 08:56:16 - mmengine - INFO - Distributed training is not used, all SyncBatchNorm (SyncBN) layers in the model will be automatically reverted to BatchNormXd layers if they are used.
/home/yuheng/miniconda3/envs/Trident/lib/python3.10/site-packages/mmseg/engine/hooks/visualization_hook.py:60: UserWarning: The draw is False, it means that the hook for visualization will not take effect. The results will NOT be visualized or stored.
  warnings.warn('The draw is False, it means that the '
11/22 08:56:16 - mmengine - INFO - Hooks will be executed in the following order:
before_run:
(VERY_HIGH   ) RuntimeInfoHook                    
(BELOW_NORMAL) LoggerHook                         
 -------------------- 
before_train:
(VERY_HIGH   ) RuntimeInfoHook                    
(NORMAL      ) IterTimerHook                      
(VERY_LOW    ) CheckpointHook                     
 -------------------- 
before_train_epoch:
(VERY_HIGH   ) RuntimeInfoHook                    
(NORMAL      ) IterTimerHook                      
(NORMAL      ) DistSamplerSeedHook                
 -------------------- 
before_train_iter:
(VERY_HIGH   ) RuntimeInfoHook                    
(NORMAL      ) IterTimerHook                      
 -------------------- 
after_train_iter:
(VERY_HIGH   ) RuntimeInfoHook                    
(NORMAL      ) IterTimerHook                      
(NORMAL      ) SegVisualizationHook               
(BELOW_NORMAL) LoggerHook                         
(LOW         ) ParamSchedulerHook                 
(VERY_LOW    ) CheckpointHook                     
 -------------------- 
after_train_epoch:
(NORMAL      ) IterTimerHook                      
(LOW         ) ParamSchedulerHook                 
(VERY_LOW    ) CheckpointHook                     
 -------------------- 
before_val:
(VERY_HIGH   ) RuntimeInfoHook                    
 -------------------- 
before_val_epoch:
(NORMAL      ) IterTimerHook                      
 -------------------- 
before_val_iter:
(NORMAL      ) IterTimerHook                      
 -------------------- 
after_val_iter:
(NORMAL      ) IterTimerHook                      
(NORMAL      ) SegVisualizationHook               
(BELOW_NORMAL) LoggerHook                         
 -------------------- 
after_val_epoch:
(VERY_HIGH   ) RuntimeInfoHook                    
(NORMAL      ) IterTimerHook                      
(BELOW_NORMAL) LoggerHook                         
(LOW         ) ParamSchedulerHook                 
(VERY_LOW    ) CheckpointHook                     
 -------------------- 
after_val:
(VERY_HIGH   ) RuntimeInfoHook                    
 -------------------- 
after_train:
(VERY_HIGH   ) RuntimeInfoHook                    
(VERY_LOW    ) CheckpointHook                     
 -------------------- 
before_test:
(VERY_HIGH   ) RuntimeInfoHook                    
 -------------------- 
before_test_epoch:
(NORMAL      ) IterTimerHook                      
 -------------------- 
before_test_iter:
(NORMAL      ) IterTimerHook                      
 -------------------- 
after_test_iter:
(NORMAL      ) IterTimerHook                      
(NORMAL      ) SegVisualizationHook               
(BELOW_NORMAL) LoggerHook                         
 -------------------- 
after_test_epoch:
(VERY_HIGH   ) RuntimeInfoHook                    
(NORMAL      ) IterTimerHook                      
(BELOW_NORMAL) LoggerHook                         
 -------------------- 
after_test:
(VERY_HIGH   ) RuntimeInfoHook                    
 -------------------- 
after_run:
(BELOW_NORMAL) LoggerHook                         
 -------------------- 
11/22 08:56:16 - mmengine - WARNING - The prefix is not set in metric class IoUMetric.
11/22 08:56:23 - mmengine - INFO - Iter(test) [  50/1449]    eta: 0:02:57  time: 0.1244  data_time: 0.0018  memory: 3559  
11/22 08:56:28 - mmengine - INFO - Iter(test) [ 100/1449]    eta: 0:02:41  time: 0.1041  data_time: 0.0012  memory: 3493  
11/22 08:56:34 - mmengine - INFO - Iter(test) [ 150/1449]    eta: 0:02:29  time: 0.1076  data_time: 0.0013  memory: 2393  
11/22 08:56:39 - mmengine - INFO - Iter(test) [ 200/1449]    eta: 0:02:20  time: 0.1070  data_time: 0.0014  memory: 3585  
11/22 08:56:44 - mmengine - INFO - Iter(test) [ 250/1449]    eta: 0:02:12  time: 0.1040  data_time: 0.0014  memory: 3690  
11/22 08:56:49 - mmengine - INFO - Iter(test) [ 300/1449]    eta: 0:02:05  time: 0.1021  data_time: 0.0013  memory: 2585  
11/22 08:56:54 - mmengine - INFO - Iter(test) [ 350/1449]    eta: 0:01:59  time: 0.1032  data_time: 0.0013  memory: 2686  
11/22 08:57:00 - mmengine - INFO - Iter(test) [ 400/1449]    eta: 0:01:53  time: 0.1061  data_time: 0.0014  memory: 2384  
11/22 08:57:05 - mmengine - INFO - Iter(test) [ 450/1449]    eta: 0:01:47  time: 0.1069  data_time: 0.0016  memory: 2790  
11/22 08:57:10 - mmengine - INFO - Iter(test) [ 500/1449]    eta: 0:01:42  time: 0.1141  data_time: 0.0017  memory: 2396  
11/22 08:57:16 - mmengine - INFO - Iter(test) [ 550/1449]    eta: 0:01:36  time: 0.1035  data_time: 0.0014  memory: 2384  
11/22 08:57:21 - mmengine - INFO - Iter(test) [ 600/1449]    eta: 0:01:31  time: 0.1078  data_time: 0.0013  memory: 2384  
11/22 08:57:27 - mmengine - INFO - Iter(test) [ 650/1449]    eta: 0:01:26  time: 0.1196  data_time: 0.0017  memory: 3717  
11/22 08:57:32 - mmengine - INFO - Iter(test) [ 700/1449]    eta: 0:01:21  time: 0.1010  data_time: 0.0015  memory: 3664  
11/22 08:57:37 - mmengine - INFO - Iter(test) [ 750/1449]    eta: 0:01:15  time: 0.1049  data_time: 0.0014  memory: 2509  
11/22 08:57:43 - mmengine - INFO - Iter(test) [ 800/1449]    eta: 0:01:10  time: 0.1152  data_time: 0.0018  memory: 2509  
11/22 08:57:49 - mmengine - INFO - Iter(test) [ 850/1449]    eta: 0:01:04  time: 0.1054  data_time: 0.0012  memory: 2508  
11/22 08:57:54 - mmengine - INFO - Iter(test) [ 900/1449]    eta: 0:00:59  time: 0.1041  data_time: 0.0012  memory: 2383  
11/22 08:57:59 - mmengine - INFO - Iter(test) [ 950/1449]    eta: 0:00:53  time: 0.0994  data_time: 0.0011  memory: 2382  
11/22 08:58:04 - mmengine - INFO - Iter(test) [1000/1449]    eta: 0:00:48  time: 0.0989  data_time: 0.0011  memory: 2489  
11/22 08:58:09 - mmengine - INFO - Iter(test) [1050/1449]    eta: 0:00:42  time: 0.1046  data_time: 0.0011  memory: 2382  
11/22 08:58:14 - mmengine - INFO - Iter(test) [1100/1449]    eta: 0:00:37  time: 0.1040  data_time: 0.0011  memory: 2388  
11/22 08:58:20 - mmengine - INFO - Iter(test) [1150/1449]    eta: 0:00:32  time: 0.1027  data_time: 0.0011  memory: 2509  
11/22 08:58:25 - mmengine - INFO - Iter(test) [1200/1449]    eta: 0:00:26  time: 0.1041  data_time: 0.0011  memory: 2508  
11/22 08:58:31 - mmengine - INFO - Iter(test) [1250/1449]    eta: 0:00:21  time: 0.1178  data_time: 0.0017  memory: 3777  
11/22 08:58:36 - mmengine - INFO - Iter(test) [1300/1449]    eta: 0:00:16  time: 0.1213  data_time: 0.0018  memory: 2384  
11/22 08:58:42 - mmengine - INFO - Iter(test) [1350/1449]    eta: 0:00:10  time: 0.1053  data_time: 0.0016  memory: 2508  
11/22 08:58:47 - mmengine - INFO - Iter(test) [1400/1449]    eta: 0:00:05  time: 0.1042  data_time: 0.0014  memory: 2508  
11/22 08:58:52 - mmengine - INFO - per class results:
11/22 08:58:52 - mmengine - INFO - 
+-------------+-------+-------+
|    Class    |  IoU  |  Acc  |
+-------------+-------+-------+
|  background | 86.92 | 94.59 |
|  aeroplane  |  70.3 | 90.21 |
|   bicycle   | 39.17 | 98.62 |
|     bird    | 87.77 | 90.09 |
|     boat    | 67.22 | 79.21 |
|    bottle   | 76.48 | 93.23 |
|     bus     | 86.39 | 97.67 |
|     car     | 80.61 | 98.04 |
|     cat     | 92.76 | 97.02 |
|    chair    | 38.02 | 47.18 |
|     cow     | 91.57 | 95.99 |
| diningtable | 47.53 |  79.8 |
|     dog     | 92.15 | 95.44 |
|    horse    | 70.76 | 97.61 |
|  motorbike  | 73.18 | 83.26 |
|    person   |  4.65 |  4.65 |
| pottedplant | 40.62 | 77.76 |
|    sheep    |  78.8 | 99.21 |
|     sofa    | 57.96 | 82.56 |
|    train    | 79.94 | 91.64 |
|  tvmonitor  |  46.2 | 50.43 |
+-------------+-------+-------+
11/22 08:58:52 - mmengine - INFO - Iter(test) [1449/1449]    aAcc: 88.6000  mIoU: 67.1000  mAcc: 83.0600  data_time: 0.0015  time: 0.1077

**Comment by zhchen9833** [comment]
Thank you for your response. Perhaps I made a mistake with the weight. I will try again. Thanks for your interesting work.

**Comment by QingXiaoYan** [comment]
> 为什么我复现后的结果比论文中的低？例如，VOC21 的 mIOU 为 65.29，低于论文中的 67.1。 我没有更改项目中的任何其他参数，只是按照您提供的步骤操作，例如“python eval.py --config config/cfd_VOC21.py --workdir output/VOC21 --sam_define”。我是否需要调整任何参数？

Could you please tell me if you were able to reproduce the results in the paper? The results I obtained by reproducing are the same as yours.
